unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/BuilderMode/BIWContext.cs

```diff
@@ -0,0 +1,82 @@
+    private const string INPUTS_PATH = "ScriptableObjects/InputReferences";
+
+    //Scriptable Objects
+    public BIWProjectReferences projectReferences => projectReferencesValue;
+    public BIWGodModeDynamicVariables godModeDynamicVariables => godModeDynamicVariablesValue;
+    public BIWFirstPersonDynamicVariables firstPersonDynamicVariables => firstPersonDynamicVariablesValue;
+    public BIWInputsReferences inputsReferences => inputsReferencesValue;
+
+    private BIWProjectReferences projectReferencesValue;
+    private BIWGodModeDynamicVariables godModeDynamicVariablesValue;
+    private BIWFirstPersonDynamicVariables firstPersonDynamicVariablesValue;
+    private BIWInputsReferences inputsReferencesValue;
+
+    // BIW Controllers
+    public IBIWOutlinerController outlinerController => outlinerControllerValue;
+    public IBIWInputHandler inputHandler => inputHandlerValue;
+    public IBIWInputWrapper inputWrapper => inputWrapperValue;
+    public IBIWPublishController publishController => publishControllerValue;
+    public IBIWCreatorController creatorController => creatorControllerValue;
+    public IBIWModeController modeController => modeControllerValue;
+    public IBIWFloorHandler floorHandler => floorHandlerValue;
+    public IBIWEntityHandler entityHandler => entityHandlerValue;
+    public IBIWActionController actionController => actionControllerValue;
+    public IBIWSaveController saveController => saveControllerValue;
+
+    private IBIWOutlinerController outlinerControllerValue;
+    private IBIWInputHandler inputHandlerValue;
+    private IBIWInputWrapper inputWrapperValue;
+    private IBIWPublishController publishControllerValue;
+    private IBIWCreatorController creatorControllerValue;
+    private IBIWModeController modeControllerValue;
+    private IBIWFloorHandler floorHandlerValue;
+    private IBIWEntityHandler entityHandlerValue;
+    private IBIWActionController actionControllerValue;
+    private IBIWSaveController saveControllerValue;
+
+    public void Init(IBIWOutlinerController outlinerController,
+        IBIWInputHandler bIWInputHandler,
+        IBIWInputWrapper biwInputWrapper,
+        IBIWPublishController biwPublishController,
+        IBIWCreatorController biwCreatorController,
+        IBIWModeController biwModeController,
+        IBIWFloorHandler biwFloorHandler,
+        IBIWEntityHandler biwEntityHandler,
+        IBIWActionController ibiwActionController,
+        IBIWSaveController biwSaveController)
+    {
+        projectReferencesValue = Resources.Load<BIWProjectReferences>(PROJECT_REFERENCES_PATH);
+        godModeDynamicVariablesValue = Resources.Load<BIWGodModeDynamicVariables>(GOD_MODE_DYNAMIC_VARIABLE_PATH);
+        firstPersonDynamicVariablesValue = Resources.Load<BIWFirstPersonDynamicVariables>(FIRST_PERSON_DYNAMIC_VARIABLE_PATH);
+        inputsReferencesValue = Resources.Load<BIWInputsReferences>(INPUTS_PATH);
```

Comment:
Maybe we can add the `asset` suffix. So it's more clear that they must be freed.

i.e.: `projectReferencesAsset`

Also, we shouldn't be calling `Instantiate` on these Assets? Because if we don't do so, modifying them in runtime will modify the assets in the project and pollute the git changelist.

Reply:
Changed the name.

For the Instantiate, it is not necessary, if you made a change in runtime to the asset, it will be reverted after you exit from play mode